crates/turbopack-ecmascript/src/lib.rs

```diff
@@ -741,7 +799,8 @@ async fn gen_content_with_visitors(
         Ok(EcmascriptModuleContent {
             inner_code: bytes.into(),
             source_map: Some(Vc::upcast(srcmap)),
-            is_esm: eval_context.is_esm(),
+            is_esm: eval_context.is_esm()
+                || specified_module_type == SpecifiedModuleType::EcmaScript,
```

Comment:
not sure if this is the best way to pass the `SpecifiedModuleType` into this, maybe it should be added to the `eval_context`? 🤔

Reply:
Yep, that would be better I guess

Reply:
But then we need to pass in information at parse time which isn't needed to parse 🤔

Reply:
Don't we need `specified_module_type` at parse time to determine the exports type in the case when no commonjs is detected.